### ISSUE TITLE
'Error: Cannot open audio file' when testing Multiple Files using GUI

### ISSUE THREAD

**Issue by tombradferlawrence** [kahst/BirdNET-Analyzer#172]
Hello,
Thanks for all your work on BirdNET and the GUI. I'm just getting started with training a custom model and running into problems. I can train a classifier ok, but when trying to test it on multiple files, it fails for most of the recordings. By which I mean, I get the error message in the title, followed by the path and filename of the recording: "Error: Cannot open audio file C:\Users\Tom\Folder\Recording.wav". Then once it has finished running, it states "false" in the "Execution" column beside the name of the unopened recording and I don't get a results file.

However, it doesn't fail for _all_ recordings; in my test set of 11, it will run on 2 of them. Confusingly, it is not always the same 2 files. Each time I run the test, a different pair are analysed successfully. So, I don't think it's a permissions issue or something like that. Recordings are all .wav format, ~2 hrs 50 mins long, originally with a 32k sample rate, the names are solely alphanumeric characters.

Am I doing something daft? Can anyone suggest what I'm doing wrong?

Thanks

**Comment by panchorivasf** [comment]
I have the same issue. Initially, I thought it was because I was trying to access a network drive. I copied the folder into a local drive and it didn't work. My next thought was the number of files (1-minute recording every 10 minutes for 4 months), so I organized the recording in folders per month. I was able to parse the first two months were analyzed, but then the issue came back. I don't know what is causing it because the error message seems generic. I'm using stereo files with a 48 kHz sample rate and 16-bit depth.

**Comment by panchorivasf** [comment]
UPDATE.
@tombradferlawrence
In my case, it seems that the error was corrupted files. Ben Gottesman (Cornell Lab) told me that this might be the issue so I went to the folder and sorted the files out by size. I moved out the files that were abnormally small and it worked.  If your situation is the same, perhaps the fact that there were different recordings that triggered the error might be due to parallel processing. Were you using multiple threads?

**Comment by Josef-Haupt** [comment]
Hi thank you for the update!
Could you provide us with a faulty file, for testing purposes?

**Comment by tombradferlawrence** [comment]
Hi @panchorivasf, glad to hear that you have been able to resolve your issue. Unfortunately, I don't think that is the case with my situation. My files are all the same size, and I've used them for other tasks without any problems, (I can load them in R and Audacity etc.) I wondered if it might be because they are quite large? Each file is >600 Mb @Josef-Haupt could that be a problem?

**Comment by popkowiersma** [comment]
I might have a similar problem: I have files of c. 2.5 GB (wave, 16 bit, 32kHz). I use the BirdNET Windows execute file, calling it from a batch file. For most of the files I receive errors (cannot read file). I process with multiple threads (2-4). I cannot detect a pattern in which files work and which don't. I record 1-hour files (to not loose a whole night when batteries run out) with Audiomoth v. 1.2. I merge them into one-night files in Goldwave, which is more practical in Raven.

**Comment by KalmiaLatifolia** [comment]
I had a similar problem which I managed to resolve.
Some batches of files would run fine through the GUI, while others would generate an error and not run.

The issue turned out to be dot underscore files. Files that start with "._" hold metadata and are generated when you open a file on a mac system (as far as I understand). Windows hides these files unless you expressly change the settings to make them visible. I had to change the view settings on my computer, find all the "._" files hidden in my folders, delete them, and then the BirdNET GUI multiple file would run.

This is what the error looked like before it was resolved:

Traceback (most recent call last):
  File "librosa\core\audio.py", line 743, in get_duration
  File "soundfile.py", line 467, in info
  File "soundfile.py", line 412, in __init__
  File "soundfile.py", line 658, in __init__
  File "soundfile.py", line 1216, in _open
soundfile.LibsndfileError: Error opening 'E:\\Massaha-new-PA\\TBNI21\\20210719_Schedule [+00.6856+013.2189]\\._S20210719T055952781+0100_E20210719T062947833+0100_+00.6856+013.2189.wav': Format not recognised.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "gradio\routes.py", line 394, in run_predict
  File "gradio\blocks.py", line 1075, in process_api
  File "gradio\blocks.py", line 884, in call_function
  File "anyio\to_thread.py", line 31, in run_sync
  File "anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
  File "anyio\_backends\_asyncio.py", line 867, in run
  File "gui.py", line 756, in select_directory_on_empty
  File "gui.py", line 427, in select_directory
  File "gui.py", line 428, in <listcomp>
  File "librosa\util\decorators.py", line 88, in inner_f
  File "librosa\core\audio.py", line 745, in get_duration
  File "audioread\__init__.py", line 132, in audio_open
audioread.exceptions.NoBackendError
